script/splits/DeployScript.s.sol

```diff
@@ -0,0 +1,146 @@
+    if (privKey == 0) {
+      console.log("PRIVATE_KEY is not set");
+      return;
+    }
+    deployerAddress = vm.addr(privKey);
+
+    require(pullSplitFactory != address(0), "PullSplitFactory address is not set");
+    require(pushSplitFactory != address(0), "PushSplitFactory address is not set");
+    pullFactory = ISplitFactoryV2(pullSplitFactory);
+    pushFactory = ISplitFactoryV2(pushSplitFactory);
+
+    console.log("Reading splits configuration from file: %s", splitsConfigFilePath);
+    SplitConfig[] memory configSplits = readSplitsConfig(splitsConfigFilePath);
+    require(configSplits.length > 0, "No splits found in the configuration file.");
+    console.log("Found %d splits in the configuration file", configSplits.length);
+
+    for (uint256 i = 0; i < configSplits.length; i++) {
+      indices[configSplits[i].name] = i;
+    }
+
+    vm.startBroadcast(privKey);
+
+    for (uint256 i = 0; i < configSplits.length; i++) {
+      deploySplit(configSplits, configSplits[i].name);
+    }
+
+    writeDeploymentJson(getFileName(splitsConfigFilePath), configSplits);
+
+    vm.stopBroadcast();
+  }
+
+  function deploySplit(SplitConfig[] memory configSplits, string memory splitName) public returns (address) {
+    if (deployments[splitName] != address(0) && deployments[splitName] != DEPLOYING_SPLIT) {
+      return deployments[splitName];
+    }
+
+    if (deployments[splitName] == DEPLOYING_SPLIT) {
+      console.log("Split %s is already processing, it must be a loop", splitName);
+      revert("Loop detected in splits configuration.");
+    }
+    deployments[splitName] = DEPLOYING_SPLIT;
+
+    SplitConfig memory split = configSplits[indices[splitName]];
+
+    address[] memory recipients = new address[](split.allocations.length);
+    uint256[] memory allocations = new uint256[](split.allocations.length);
+
+    for (uint256 j = 0; j < split.allocations.length; j++) {
+      if (LibString.startsWith(split.allocations[j].recipient, "0x")) {
+        recipients[j] = vm.parseAddress(split.allocations[j].recipient);
+      } else if (bytes(split.allocations[j].recipient).length > 0) {
+        recipients[j] = deploySplit(configSplits, split.allocations[j].recipient);
+      } else {
+        console.log("Recipient address is not set for allocation %d in split %s", j, splitName);
+        revert("Recipient address is not set for allocation.");
+      }
+
+      allocations[j] = split.allocations[j].allocation;
+    }
+
+    ISplitFactoryV2.Split memory newSplit = ISplitFactoryV2.Split({
+      recipients: recipients,
+      allocations: allocations,
+      totalAllocation: split.totalAllocation,
+      distributionIncentive: uint16(split.distributionIncentive)
+    });
+
+    address splitAddress;
+    if (compareStrings(split.splitType, "pull")) {
+      splitAddress = pullFactory.createSplit(newSplit, split.owner, deployerAddress);
+    } else if (compareStrings(split.splitType, "push")) {
```

Comment:
In what cases will we use push splits? (out of interest)

Reply:
I just supported both available types without having on my mind a specific use case. Maybe that would be a squad/NOPs preference?